tests/unit/states/test_file.py

```diff
@@ -1255,30 +1262,91 @@ def test_directory(self):
                                                  (name, user=user, group=group),
                                                  ret)
 
+                        if salt.utils.platform.is_windows():
+                            isdir_side_effect = [False, True, False]
+                        else:
+                            isdir_side_effect = [True, False, True, False]
                         with patch.object(os.path, 'isdir',
-                                          MagicMock(side_effect=[True, False, True, True])):
+                                          MagicMock(side_effect=isdir_side_effect)):
                             comt = ('Failed to create directory {0}'
                                     .format(name))
-                            ret.update({'comment': comt, 'result': False})
+                            ret.update({'comment': comt,
+                                             'result': False,
+                                             'changes': {name: 'New Dir'}})
                             self.assertDictEqual(filestate.directory
                                                  (name, user=user, group=group),
                                                  ret)
 
+                        check_perms_ret = {'name': name,
+                                           'result': False,
+                                           'comment': '',
+                                           'changes': {}}
+                        if salt.utils.platform.is_windows():
+                            mock_perms = MagicMock(return_value=check_perms_ret)
+                        else:
+                            mock_perms = MagicMock(return_value=(check_perms_ret, ''))
+
+                        recurse = ['silent']
+                        ret = {'name': name,
+                               'result': False,
+                               'comment': '',
+                               'pchanges': 'Changes silenced',
```

Comment:
should remove pchanges here as well in favor of changes

Reply:
Also done!